app/components/HeaderNotifications/index.js

```diff
@@ -100,12 +100,20 @@ export default class NotificationsDropdown extends Component<Props, State> {
         toggle={() =>
           this.setState(
             {
-              notificationsOpen: !this.state.notificationsOpen
+              notificationsOpen: !this.state.notificationsOpen,
+              hideUnreadCount: true
```

Comment:
I don't think you need to keep this in state? `markAllNotifications` should update the state in redux - if you want it to update right away (and e.g. not wait until the response comes back) you can add that manually to the `notificationsFeed` reducer on `NotificationsFeed.MARK_ALL.SUCCESS`